src/find/matchers/time.rs

```diff
@@ -362,4 +504,155 @@ mod tests {
+                // this has just been created, so should be newer
+                File::create(temp_dir.path().join(new_file_name)).expect("create temp file");
+                let new_file = get_dir_entry_for(&temp_dir_path, new_file_name);
+                // this file should already exist
+                let old_file = get_dir_entry_for("test_data", "simple");
+                let deps = FakeDependencies::new();
+                let matcher = NewerOptionMatcher::new(
+                    x_option.to_string(),
+                    y_option.to_string(),
+                    &old_file.path().to_string_lossy(),
+                );
+
+                assert!(
+                    matcher
+                        .unwrap()
+                        .matches(&new_file, &mut deps.new_matcher_io()),
+                    "new_file should be newer than old_dir"
+                );
+
+                // After the file is deleted, DirEntry will point to an empty file location,
+                // thus causing the Matcher to generate an IO error after matching.
+                let _ = fs::remove_file(&*new_file.path().to_string_lossy());
+                let matcher = NewerOptionMatcher::new(
+                    x_option.to_string(),
+                    y_option.to_string(),
+                    &old_file.path().to_string_lossy(),
+                );
+                assert!(
+                    !matcher
+                        .unwrap()
+                        .matches(&new_file, &mut deps.new_matcher_io()),
+                    "The correct situation is that the file reading here cannot be successful."
+                );
+            }
+        }
+    }
+
+    #[test]
+    fn newer_time_matcher() {
+        let deps = FakeDependencies::new();
+        let time = deps
+            .new_matcher_io()
+            .now()
+            .duration_since(SystemTime::UNIX_EPOCH)
+            .unwrap()
+            .as_millis()
+            .try_into()
+            .unwrap();
+
+        let created_matcher = NewerTimeMatcher::new(NewerOptionType::Birthed, time);
+
+        thread::sleep(Duration::from_secs(2));
+        let temp_dir = Builder::new()
+            .prefix("newer_time_matcher")
+            .tempdir()
+            .unwrap();
+        // No easy way to independently set file times. So create it - setting creation time
+        let foo_path = temp_dir.path().join("foo");
+        // after "time" created a file
+        let _ = File::create(&foo_path).expect("create temp file");
+        // so this file created time should after "time"
+        let file_info = get_dir_entry_for(&temp_dir.path().to_string_lossy(), "foo");
+        assert!(
+            created_matcher.matches(&file_info, &mut deps.new_matcher_io()),
+            "file created time should after 'time'"
+        );
+
+        // accessed time test
+        let accessed_matcher = NewerTimeMatcher::new(NewerOptionType::Accessed, time);
+
+        thread::sleep(Duration::from_secs(2));
```

Comment:
why sleep ?

Reply:
This is my oversight, sleep() is not really needed here, and there are two other codes with the same problem. They have been modified in the commit. Thanks!